### PR TITLE
feat: Rename programinstance DB table to enrollment [ TECH-1593 ]

### PR DESCRIPTION
**Pull request by lucaCambi77**
Rename `programinstance` to `enrollment` from DB to application code
see https://github.com/dhis2/dhis2-core/pull/14193

Notes:

- `enrollment_messageconversation` and `event_messageconversation` are no longer in use and will be removed. We need to create an issue for that
- on the Sierra DB there is a `programinstance_outboundsms` but I can't really find a trace of this table not even in migrations (probably a very old table) so we don't take action about that

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcTeiEnrollmentsAnalyticsTableManager.java [hunk 1]
```diff
@@ -238,7 +238,7 @@ protected void populateTable(
     }
 
     removeLastComma(sql)
-        .append(" from programinstance pi")
+        .append(" from enrollment pi")
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcTeiEnrollmentsAnalyticsTableManager.java:241]: When do you plan to rename aliases? We should probably settle on new ones, maybe `en` for `enrollment`, `ev` for `event` and `te` for `trackedentity` 🤔 

@enricocolasante

**Review by enricocolasante — APPROVED** [review]
Can you already create issues for removing the unused tables and link them to the renaming epic?

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcTeiEnrollmentsAnalyticsTableManager.java (hunk 1 above)
Comment (lucaCambi77) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcTeiEnrollmentsAnalyticsTableManager.java:241]: @teleivo @enricocolasante 
https://dhis2.atlassian.net/browse/TECH-1596 https://dhis2.atlassian.net/browse/TECH-1597

**Review by maikelarabori — APPROVED** [review]
Looks good, just a minor comment related to the SQL file (lowercase letters) 👍

File: dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.41/V2_41_1000__ProgramInstanceToEnrollment.sql [hunk 2]
```diff
@@ -0,0 +1,86 @@
+-- rename programstageinstance to event
+alter table if exists programinstance rename to enrollment;
```
[the remark this answers is not in the document]
Comment (lucaCambi77) [reply dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.41/V2_41_1000__ProgramInstanceToEnrollment.sql]: done

**Pull request merged**